pyiceberg/expressions/__init__.py

```diff
@@ -52,8 +52,14 @@
     ConfigDict = dict
 
 
-def _to_unbound_term(term: Union[str, UnboundTerm[Any]]) -> UnboundTerm[Any]:
-    return Reference(term) if isinstance(term, str) else term
+def _to_unbound_term(term: Union[str, UnboundTerm[Any], BoundReference[Any]]) -> UnboundTerm[Any]:
+    if isinstance(term, str):
+        return Reference(term)
+    if isinstance(term, UnboundTerm):
+        return term
+    if isinstance(term, BoundReference):
+        return Reference(term.field.name)
+    raise ValueError(f"Expected UnboundTerm | BoundReference | str, got {type(term).__name__}")
```

Comment:
There were some faulty tests: https://github.com/apache/iceberg-python/pull/2632/

```suggestion
def _to_unbound_term(term: Union[str, UnboundTerm[Any]]) -> UnboundTerm[Any]:
    return Reference(term) if isinstance(term, str) else term
```

Reply:
That PR makes sense, thanks for doing it. Now the LiteralPredicate makes more sense to me.